### ISSUE TITLE
[MSIT][Testing] Gauge test infinite loop

### ISSUE THREAD

**Issue by schmerl** [cmu-able/rainbow#66]
If the gauge does not issue an operation (for example, because it throws an exception) then the test never finishes. Tests should always finish.

**Comment by schmerl** [comment]
So, this prevents writing a test that ensures that if I do not send the right information from the probe, that no operation is produced. For example, the captcha gauge takes two possible probe values "on" or "off". If I write this test case:

```java
   	Map<String,IRainbowOperation> mappings = new HashMap<>();
    	mappings.putAll(gd.mappings());
    	CaptchaGauge cg = new CaptchaGauge(gd.gaugeName(), 10000L, new TypedAttribute(gd.gaugeName(),gd.gaugeType()), gd.modelDesc(), gd.setupParams(), mappings);
    	cg.initialize(new LoggerRainbowReportingPort());
    	cg.start();
    	
    	cg.reportFromProbe(mockProbeIdentifier("mocked", "testing", "testing"), "foor");
    	
    	IRainbowOperation operation = GaugeTestingUtil.waitForNextOperation();
    	assertNull(operation);
```
The test neither succeeds or fails.

**Comment by IndeedPlusPlus** [comment]
The designed preferred way to set a timeout for waiting is to put the timeout on the test case.
```java
@Test(timeout = 3000L)
public void someTestWithInfinityWait() throws Exception {
// ...
GaugeTestingUtil.waitForNextOperation();
}
```

We will update the document to reflect this design.

**Comment by schmerl** [comment]
This doesn't work the way I expect it to for this test case. If the test times out with timeout=3000L it throws an exception and the testcase fails. In this case here, I want it to pass (the operation should be null).

**Comment by IndeedPlusPlus** [comment]
Noted. We will add that to our issue tracker.

**Comment by IndeedPlusPlus** [comment]
We've added a timeout so that we can assert that no operation/output occurs during the given timeout.

**Comment by IndeedPlusPlus** [comment]
fixed in #70
